build_runner/lib/src/asset/reader.dart

```diff
@@ -29,7 +30,7 @@ abstract class RunnerAssetReader implements MultiPackageAssetReader {}
 /// tracking.
 class SingleStepReader implements AssetReader {
   final AssetGraph _assetGraph;
-  final _assetsRead = new Set<AssetId>();
+  final _assetsRead = new SplayTreeSet<AssetId>();
```

Comment:
nit: we should add a comment here about ordering so this doesn't get reverted

Reply:
Yes, this is the more important part if anything imo.

Reply:
https://github.com/dart-lang/build/pull/1327